packages/react-ui-core/src/Gmap/utils/markerHelpers.js

```diff
@@ -0,0 +1,36 @@
+import omitBy from 'lodash/omitBy'
+import { setupEvents } from './mapEventHelpers'
+
+const EVENTS = {
+  onClick: 'click',
+  onDoubleClick: 'dblclick',
+  onDragEnd: 'dragend',
+  onMouseDown: 'mousedown',
+  onMouseOut: 'mouseout',
+  onMouseOver: 'mouseover',
+  onMouseUp: 'mouseup',
+  onRecenter: 'recenter',
+}
+
+export const removeMarker = marker => {
+  if (marker) {
+    window.google.maps.event.clearInstanceListeners(marker)
+    marker.setMap(null)
+  }
+}
+
+export const setupMarker = (map, props) => {
+  if (map) {
+    const propsWithoutEvents = omitBy(props, (key, val) => EVENTS[val])
+    const marker = new window.google.maps.Marker({
+      ...propsWithoutEvents,
```

Comment:
for my curiosity, why does a marker not get events?

Reply:
It does, but they have to be added after the marker has been instantiated. We could probably leave it "as is" and pass the events through as well as I don't think it would hurt anything.

Reply:
oh no this is good. just wondering

Reply:
Yeah, I wish we could just pass them in, I don't understand why you can't. 🤷‍♂️